alibi/explainers/integrated_gradients.py

```diff
@@ -400,6 +400,14 @@ def _gradients_input(model: Union[tf.keras.models.Model],
 
     grads = tape.gradient(preds, x)
 
+    # if there are inputs have not impact to the output, the gradient is None, but we need to return a tensor
+    if isinstance(x, list):
+        shape = x[0].shape
+    else:
+        shape = x.shape
+    for idx, grad in enumerate(grads):
```

Comment:
If our input x is not a list, I think tape.gradient may directly output the gradient for x, in which case we may not want to have this enumerate step which seems to assume that the grads is a list of gradient tensors (one for each input).

Reply:
And actually, if our input x isn't a list, would we encounter the None gradients?  

It seems like this it primarily comes up for us because we have outputs y1, y2, y3 which depend on different subsets of inputs x1, x2, x3.  If y1 only depends on x1, then if we try to explain the model, we can run into issues because the gradients for x2 and x3 will be none.

But if the input isn't a list and is just x, then it seems like every output would need to depend on the whole input tensor?

Reply:
If that is the case, maybe we can only do this gradient zero-ing for when x is a list?  Something like:
```
if isinstance(x, list):
    for idx, grad in enumerate(grads):
        if grad is None:
            grads[idx] = tf.convert_to_tensor(np.zeros(shape), dtype=x[idx].dtype)
```